cypress/integration/chi-custom-elements/dropdown.spec.js

```diff
@@ -0,0 +1,233 @@
+
+const hasClassAssertion = (el, value) => {
+  cy.get(el).should('have.class', value);
+};
+
+describe('Dropdown', () => {
+  before(() => {
+    cy.visit('tests/custom-elements/dropdown.html');
+  });
+
+  describe.only('Base', () => {
+    beforeEach(() => {
+      cy.get(DROPDOWN_DATA_CY.BASE).find(DROPDOWN_TRIGGER).as(
+        'dropdownTrigger'
+      );
+      cy.get(DROPDOWN_DATA_CY.BASE).find(DROPDOWN_MENU).as(
+        'dropdownMenu'
+      );
+    });
+
+    it.only('Should open the dropdown', () => {
+      cy.get(`@dropdownMenu`)
+        .should('not.be.visible');
+      cy.get(`@dropdownTrigger`).click();
+      cy.get(`@dropdownMenu`)
+        .should('be.visible');
+    });
+
+    it.only('Should close the dropdown', () => {
+      cy.get(`@dropdownMenu`)
+        .should('be.visible');
+      cy.get(`@dropdownTrigger`).click();
+      cy.get(`@dropdownMenu`)
+        .should('not.be.visible');
+    });
+  });
+
+  describe.only('Animate', () => {
+    beforeEach(() => {
+      cy.get(DROPDOWN_DATA_CY.ANIMATE).find(DROPDOWN_TRIGGER).as(
+        'dropdownTrigger'
+      );
+      cy.get(DROPDOWN_DATA_CY.ANIMATE).find(DROPDOWN_MENU).as(
+        'dropdownMenu'
+      );
+    });
+
+    it('Should transform the accordion when dropdown is in active or inactive state', () => {
+      cy.get(`@dropdownTrigger`)
+        .find(`.${ACTIVE_CLASS}`)
+        .should('not.exist');
+
+      cy.get('@dropdownTrigger').click().wait(500)
+      .then($els => {
+        hasClassAssertion('@dropdownTrigger', ACTIVE_CLASS);
+        cy.get(`@dropdownMenu`).should('be.visible');
+        const win = $els[0].ownerDocument.defaultView;
+        const after = win.getComputedStyle($els[0], ':after');
+        const tr = after.transform;
+        const values = tr.split('(')[1].split(')')[0].split(',');
+        const a = values[0];
+        const b = values[1];
+        const angle = Math.round(Math.atan2(b, a) * (180/Math.PI));
+        expect(angle).to.equal(180);
+      });
+    });
+  });
+
+  describe.only('Position', () => {
+
+    DROPDOWN_DATA_CY.POSITION.forEach((position) => {
+      it.only(`Should be placed to the position ${position.placement}`, () => {
+        cy.get(position.selector).click();
+        cy.get(position.selector).find(DROPDOWN_MENU)
+          .scrollIntoView({ offset: { top: -150, left: 0 } })
+          .should('have.attr', 'style')
+          .should('contain', `transform: ${position.transform}`);
+      })
+    });
+  });
+
+  describe.only('Methods', () => {
+    describe('Toggle', () => {
+      beforeEach(() => {
+        cy.get(DROPDOWN_DATA_CY.METHOD.TOGGLE.TRIGGER).as(
+          'dropdownTrigger'
+        );
+        cy.get(DROPDOWN_DATA_CY.METHOD.TOGGLE.DROPDOWN_TOGGLE).find(DROPDOWN_MENU).as(
+          'dropdownMenu'
+        );
+      });
+      it('Should toggle the dropdown', () => {
+        cy.get(`@dropdownMenu`)
+          .should('not.be.visible');
+        cy.get(`@dropdownTrigger`).click();
```

Comment:
How does this check if the methods work? Seems like the implementation checks visibility toggling when clicking the standard trigger. The method is not called externally.

Reply:
Updated